### DISCUSSION TITLE
Trouble accessing cookies in middleware

### DISCUSSION THREAD

**Discussion by patrickyan** [vercel/next.js#42464] [Help] [answered]
I am following the method shown in the [documentation on middleware](https://nextjs.org/docs/advanced-features/middleware#using-cookies).

In my middleware:

```
export async function middleware(request: NextRequest) {
	console.log('COOKIES:', request.cookies);
	console.log('HAS XSRF-Token:', request.cookies.has('XSRF-Token'));
	const xsrf_cookie = request.cookies.get('XSRF-Token')?.value;
	console.log('XSRF-Token:', xsrf_cookie);
	return NextResponse.next();
}
```

The console output:
```
COOKIES: {
  XSRF-Token: 'XSRF-Token=aaaaaa; Path=/',
  session: 'session=aaaaaa; Path=/'
}
HAS XSRF-Token: true
XSRF-Token: undefined
```

As you can see, the cookie is clearly there, yet `get()` results in undefined

**Comment by icyJoseph** [comment, marked as answer]
Works for me, what `next` version are you using? what happens if you don't invoke `?.value`?

## Edit

OP is using Next 12.

**Comment by patrickyan** [comment]
Version 12.3.1

Removing `?.value` and just using `request.cookies.get('XSRF-Token')` seems to work correctly.

**Reply by icyJoseph** [reply]
Yeah, the value thing is a re-work done in Next 13. You need to check the Next 12 docs. I'm not sure how to see them though.

**Reply by icyJoseph** [reply]
Here ya go https://nextjs.org/docs/tag/v12.3.1/advanced-features/middleware

**Reply by icyJoseph** [reply]
As usual please don't forget to mark as answered.

**Reply by patrickyan** [reply]
I see. I know 13 was announced last week, but I didn't realize the docs would be out of date with no warning. There is zero mention of version number or incompatibility on the entire page. There should be some kind of banner/alert and explicit text saying "this was just changed in the update to 13."

**Reply by icyJoseph** [reply]
At top of the page there's a summary of changes though. I can imagine it's easy to miss.
